File: japiwznm/WznmQVisLinkctx.java
/**
	* \file WznmQVisLinkctx.java
	* Java API code for record of table TblWznmQVisLinkctx
	* \copyright (C) 2018-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 25 Aug 2024
  */
// IP header --- ABOVE

package apiwznm;

import org.w3c.dom.*;
import sbecore.*;

public class WznmQVisLinkctx {
	
	public WznmQVisLinkctx(
				int jnum
				, int orgClk
				, int dstClk
				, int Weight
				, int jnumBoxOrg
				, int jnumBoxDst
				, String Title
				, String Content1
				, String Content2
			) {
		this.jnum = jnum;
		this.orgClk = orgClk;
		this.dstClk = dstClk;
		this.Weight = Weight;
		this.jnumBoxOrg = jnumBoxOrg;
		this.jnumBoxDst = jnumBoxDst;
		this.Title = Title;
		this.Content1 = Content1;
		this.Content2 = Content2;
	};

	public int jnum;
	public int orgClk;
	public int dstClk;
	public int Weight;
	public int jnumBoxOrg;
	public int jnumBoxDst;
	public String Title;
	public String Content1;
	public String Content2;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "WznmQVisLinkctx");

		if (Xmlio.checkXPath(doc, basexpath)) {
			orgClk = Xmlio.extractIntegerUclc(doc, basexpath, "orgClk", "ocl", null, 0);
			dstClk = Xmlio.extractIntegerUclc(doc, basexpath, "dstClk", "dcl", null, 0);
			Weight = Xmlio.extractIntegerUclc(doc, basexpath, "Weight", "wgt", null, 0);
			jnumBoxOrg = Xmlio.extractIntegerUclc(doc, basexpath, "jnumBoxOrg", "", null, 0);
			jnumBoxDst = Xmlio.extractIntegerUclc(doc, basexpath, "jnumBoxDst", "", null, 0);
			Title = Xmlio.extractStringUclc(doc, basexpath, "Title", "tit", null, 0);
			Content1 = Xmlio.extractStringUclc(doc, basexpath, "Content1", "cnt1", null, 0);
			Content2 = Xmlio.extractStringUclc(doc, basexpath, "Content2", "cnt2", null, 0);
			
			return true;
		};

		return false;
	};
};

File: apiwznm/WznmQVisLinkctx.cpp
/**
	* \file WznmQVisLinkctx.cpp
	* API code for table TblWznmQVisLinkctx (implementation)
	* \copyright (C) 2016-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 25 Aug 2024
  */
// IP header --- ABOVE

#include "WznmQVisLinkctx.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WznmQVisLinkctx
 ******************************************************************************/

WznmQVisLinkctx::WznmQVisLinkctx(
			const uint jnum
			, const utinyint orgClk
			, const utinyint dstClk
			, const utinyint Weight
			, const int jnumBoxOrg
			, const int jnumBoxDst
			, const string Title
			, const string Content1
			, const string Content2
		) :
			jnum(jnum)
			, orgClk(orgClk)
			, dstClk(dstClk)
			, Weight(Weight)
			, jnumBoxOrg(jnumBoxOrg)
			, jnumBoxDst(jnumBoxDst)
			, Title(Title)
			, Content1(Content1)
			, Content2(Content2)
		{
};

bool WznmQVisLinkctx::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WznmQVisLinkctx");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractUtinyintUclc(docctx, basexpath, "orgClk", "ocl", orgClk);
		extractUtinyintUclc(docctx, basexpath, "dstClk", "dcl", dstClk);
		extractUtinyintUclc(docctx, basexpath, "Weight", "wgt", Weight);
		extractIntUclc(docctx, basexpath, "jnumBoxOrg", "", jnumBoxOrg);
		extractIntUclc(docctx, basexpath, "jnumBoxDst", "", jnumBoxDst);
		extractStringUclc(docctx, basexpath, "Title", "tit", Title);
		extractStringUclc(docctx, basexpath, "Content1", "cnt1", Content1);
		extractStringUclc(docctx, basexpath, "Content2", "cnt2", Content2);
	};

	return basefound;
};

/******************************************************************************
 class ListWznmQVisLinkctx
 ******************************************************************************/

ListWznmQVisLinkctx::ListWznmQVisLinkctx() {
};

ListWznmQVisLinkctx::ListWznmQVisLinkctx(
			const ListWznmQVisLinkctx& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQVisLinkctx(*(src.nodes[i]));
};

ListWznmQVisLinkctx& ListWznmQVisLinkctx::operator=(
			const ListWznmQVisLinkctx& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQVisLinkctx(*(src.nodes[i]));

	return *this;
};

ListWznmQVisLinkctx::~ListWznmQVisLinkctx() {
	clear();
};

void ListWznmQVisLinkctx::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWznmQVisLinkctx::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WznmQVisLinkctx* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWznmQVisLinkctx");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WznmQVisLinkctx", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WznmQVisLinkctx(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WznmQVisLinkctx[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};
